test/test_product_config.py

```diff
@@ -1198,7 +1327,7 @@ def test_defaults(self) -> None:
 class Fixture:
     """Base class providing some sample config dicts"""
 
-    def setup(self):
```

Comment:
Out of interest, why keep `setup_method` here? I had issues with going to fixtures in these situations in katdal because some test helper functions needed both the fixture and a non-fixture parameter to run (can one even mix those?).

Reply:
Good point. I've now turned each of the dictionaries in Fixture into its own pytest fixture.

Accessing fixtures from helper functions is one thing I find painful in pytest, particularly compared to the old unittest/nose style. I usually end up having to pass them into the helper function. The alternative approach is to make the helper a fixture itself (a `pytest.fixture`-decorated function that returns a function), which does let you hide the dependent fixtures, but then you have to declare the helper as a fixture parameter on every test that uses it.